src/QMCHamiltonians/ObservableHelper.cpp

```diff
@@ -147,7 +147,8 @@ void ObservableHelper::write(const value_type* first_v, const value_type* /*firs
   if (rank)
   {
     H5Sset_extent_simple(space1_id, rank, &curdims[0], &maxdims[0]);
-    H5Sselect_hyperslab(space1_id, H5S_SELECT_SET, &offsets[0], NULL, &mydims[0], NULL);
+    const std::vector<hsize_t> one(rank, 1);
```

Comment:
Document this, I just see a vector called one that does seem to have some size and is all ones. It's basically a magic number.  It's sort of explained in your PR but it should be documented here.

Reply:
> I'm not following how this isn't a breaking change for anything that reads what is being written here. Perhaps the HDF5 API is really complex enough that these two calls result in data that can be read for the hd5 file with the same call?

The net effect is the same. In the origin version, it is telling HDF5 that you are selecting mydims hyper-slabs along each dimensions with each hyper-slab of unit size 1x1x... . In this PR, it is selecting a single hyper-slab of size mydims.

In summary:
H5Sselect_hyperslab (*,*,*,*, count, NULL);
is equivalent (in terms final result) to
H5Sselect_hyperslab (*,*,*,*, one, count);
but they have different high-level meaning, and the later one is closer to what qmcpack intended to do.

Reply:
> Document this, I just see a vector called one that does seem to have some size and is all ones. It's basically a magic number. It's sort of explained in your PR but it should be documented here.

Should I put the explanation as comments before every H5Sselect_hyperslab?
I think it is intuitive given the spec of H5Sselect_hyperslab.
The vector of one means we want to write one hyper-slab (block) along each of the dimensions and each hyper-slab is of size count.
The original version write count hyper-slabs along each dimension and each hyper-slab is of size one.
They lead to the same result but literally means different objective.

Reply:
Thanks for these welcome improvements. Perhaps just comment the first usage in each file. Can also put this PR URL.